airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt

```diff
@@ -263,7 +269,7 @@ class DefaultDestinationTaskLauncher(
                 enqueue(flushCheckpointsTaskFactory.make())
             }
 
-            if (streamManager.isBatchProcessingComplete()) {
+            if (wrapped.batch.state == Batch.State.COMPLETE && streamManager.isBatchProcessingComplete()) {
```

Comment:
The first condition is wrong. Because we have groups now, it's possible for a batch to trigger completion even if its state isn't `COMPLETE`, as long as it is the last range in a group where another batch was `COMPLETE`.